src/renderer/services/maps-manager.service.ts

```diff
@@ -37,13 +37,33 @@ export class MapsManagerService {
     private readonly lastLinkedVersion$: Subject<BSVersion> = new Subject();
     private readonly lastUnlinkedVersion$: Subject<BSVersion> = new Subject();
 
+    private importListeners: ((map: BsmLocalMap, version?: BSVersion) => void)[] = [];
+    private importCount = 0;
+
     private constructor() {
         this.ipcService = IpcService.getInstance();
         this.modal = ModalService.getInstance();
         this.progressBar = ProgressBarService.getInstance();
         this.notifications = NotificationService.getInstance();
         this.config = ConfigurationService.getInstance();
         this.linker = VersionFolderLinkerService.getInstance();
+
+        this.ipcService.watch<{map: BsmLocalMap, version?: BSVersion}>("map-imported")
```

Comment:
You don't need these watchers.\
I will deprecate the watch function in the future.\
Using it too much overcomplicates the code; the backend should never send random events like this. It should only send events when a function is called, until the end of the function’s execution.

In your `importMaps`, the `"bs-maps.import-maps"` should return a `Progression<BsmLocalMap>`, and then you can do something like this:
```ts
sendV2("bs-maps.import-maps", {...args}).pipe(
  tap( progress => importListerners.foreach(listener => listener(version, progress.data))
)
```

Reply:
Got this, might just do this on `import-maps` and not on `download-maps`. I can do the refactoring on download maps once I get this PR merged.

Reply:
Let me this 👌, there is few more places that use this function, I will try to remove them all today